packages/angular/src/jsonforms.service.ts

```diff
@@ -1,58 +1,107 @@
+  
+  Copyright (c) 2017-2020 EclipseSource Munich
+  https://github.com/eclipsesource/jsonforms
+  
+  Permission is hereby granted, free of charge, to any person obtaining a copy
+  of this software and associated documentation files (the "Software"), to deal
+  in the Software without restriction, including without limitation the rights
+  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+  
+  The above copyright notice and this permission notice shall be included in
+  all copies or substantial portions of the Software.
+  
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+*/
 import {
-  configReducer,
-  CoreActions,
-  coreReducer,
-  i18nReducer,
-  JsonFormsRendererRegistryEntry,
-  JsonFormsState,
-  JsonFormsSubStates,
-  LocaleActions,
-  RankedTester,
-  setConfig,
-  SetConfigAction,
-  UISchemaActions,
-  UISchemaElement,
-  uischemaRegistryReducer,
-  ValidationMode,
-  Actions
+    Actions,
+    configReducer,
+    CoreActions,
+    coreReducer,
+    generateDefaultUISchema,
+    generateJsonSchema,
+    i18nReducer,
+    JsonFormsRendererRegistryEntry,
+    JsonFormsState,
+    JsonFormsSubStates,
+    JsonSchema,
+    LocaleActions,
+    RankedTester,
+    setConfig,
+    SetConfigAction,
+    UISchemaActions,
+    UISchemaElement,
+    uischemaRegistryReducer,
+    UISchemaTester,
+    ValidationMode
 } from '@jsonforms/core';
 import { BehaviorSubject, Observable } from 'rxjs';
 import { JsonFormsBaseRenderer } from './base.renderer';
 
 import { cloneDeep } from 'lodash';
-import { Injectable } from '@angular/core';
-
-@Injectable({
-    providedIn: 'root'
-})
 export class JsonFormsAngularService {
 
     private _state: JsonFormsSubStates;
     private state: BehaviorSubject<JsonFormsState>;
 
-    init(initialState: JsonFormsSubStates = {}) {
+    init(initialState: JsonFormsSubStates = { core: { data: undefined, schema: undefined, uischema: undefined } }) {
         this._state = initialState;
         this._state.config = configReducer(undefined, setConfig(this._state.config));
         this.state = new BehaviorSubject({ jsonforms: this._state });
+        this.updateCore(
+            Actions.init(
+                initialState.core.data,
+                initialState.core.schema ?? generateJsonSchema(initialState.core.data),
+                initialState.core.uischema ?? generateDefaultUISchema(initialState.core.schema)
```

Comment:
The generation of the ui schema needs to be done on the possibly generated schema. Otherwise the ui schema will be generated on an undefined schema when both schema and ui schema are undefined.